### ISSUE TITLE
mocha-webpack interaction with clean-webpack-plugin: output files deleted and no tests are run

### ISSUE THREAD

**Issue by github-account-because-they-want-it** [zinserjan/mocha-webpack#231]
Steps to reproduce:

1. Clone my repo __mocha-webpack branch__ [here](https://github.com/github-account-because-they-want-it/webpack-struggle/tree/mocha-webpack)
2. `npm install`
3. Run `webpack`. Files are built in the output directory as expected:
```
$ ls dist
index.html  main.64d950fb1144a8614c58.js  runtime~main.297929bbe0221ed990a6.js
```
4. Run `npm test` or `mocha-webpack`:
```
WEBPACK  Compiled with 1 warning(s)

Warning

  configuration
  The 'mode' option has not been set. Set 'mode' option to 'development' or 'production' to enable defaults for this environment.
```
Notice that the test in `test` directory is not being run.  Also all the files in the `dist` directory are being deleted, except for `index.html`:
```
$ ls dist
index.html
```
The question is, why the tests are not being run, and why the build files are deleted?

**Comment by zinserjan** [comment]
mocha-webpack 1.1.0 does not work with webpack@4. Install mocha-webpack@next

**Comment by github-account-because-they-want-it** [comment]
Thank you @zinserjan . I installed mocha-webpack@next and now the tests are running but the other issue still remains. The build output files are being deleted. I updated the repo branch above with a new commit to reflect the current status if you are interested in checking that.

Also, is it mentioned currently in the docs that **mocha-webpack@next** is needed for webpack@4?
